scripts/script_utils.py

```diff
@@ -226,3 +280,102 @@ def get_affected_files(allow_limited=True):
         result = subprocess.check_output(['git', 'ls-files'])
 
     return result.rstrip('\n').split('\n'), diff_base
+
+
+def get_pr_from_latest_commit():
+    """Attempt to determine a pull request from the latest commit.
+
+    Checks that there is only one ``#`` character in the commit subject
+    and then reads the integer that comes after.
+
+    If a valid integer is found, confirms that such a PR exists and has
+    the same ``merge_commit_sha`` as the ``HEAD`` commit.
+
+    :rtype: int
+    :returns: The ID of the pull request found in the latest commit.
+              If none can be found, returns -1.
+    """
+    commit_info = check_output(
+        'git', 'log', '-1', 'HEAD', '--pretty=%H|%s')
+    head_commit_sha, commit_subject = commit_info.split('|', 1)
+
+    # First, attempt to determine the PR ID from the commit subject.
+    matches = PR_ID_REGEX.findall(commit_subject)
+    if len(matches) != 1:
+        msg = '\n'.join([
+            'Cannot uniquely determine a commit ID from commit subject:',
+            commit_subject])
+        print(msg, file=sys.stderr)
+        return -1
+
+    # Then, use the computed ID to get PR info from GitHub.
+    id_val = int(matches[0])
+    api_url = GITHUB_API_PR_TEMPLATE % (id_val,)
+
+    response = urlopen(api_url)
```

Comment:
Also, felt very gross when doing it. Originally had `httplib2` because I assumed we'd always have those deps, but then I realized that wasn't the case.